mailbox.c

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright (c) 2012, Broadcom Europe Ltd.
```

Comment:
Can you add the link to the source in the file so that people can easily find it? Besides that, I think we are compliant by distributing in source form and keeping the copyring notice and restrictions.